lib/http1.c

```diff
@@ -707,10 +710,21 @@ void finalostream_send(h2o_ostream_t *_self, h2o_req_t *req, h2o_iovec_t *inbufs
     struct st_h2o_http1_finalostream_t *self = (void *)_self;
     struct st_h2o_http1_conn_t *conn = (struct st_h2o_http1_conn_t *)req->conn;
     h2o_iovec_t *bufs = alloca(sizeof(h2o_iovec_t) * (inbufcnt + 1));
+    int i;
     int bufcnt = 0;
 
     assert(self == &conn->_ostr_final);
 
+    /* count bytes_sent if other ostreams haven't counted */
+    if (inbufcnt > 0 && req->bytes_sent == 0) {
```

Comment:
I can understand the fact that you wanted to keep modifications within http1.c, but isn't this a bit complicated?

How about adding a flag to `h2o_req_t` that tells if the bytes are counted outside of the protocol handler (and set the flag in chunked.c)?

We already have HTTP/1 specific flag in `h2o_req_t` (as `http1_is_persistent`). I think we can add another.